libsolidity/interface/CompilerStack.cpp

```diff
@@ -841,3 +842,81 @@ string CompilerStack::computeSourceMapping(eth::AssemblyItems const& _items) con
 	}
 	return ret;
 }
+
+namespace {
+
+Json::Value gasToJson(GasEstimator::GasConsumption const& _gas)
+{
+    if (_gas.isInfinite || _gas.value > std::numeric_limits<Json::LargestUInt>::max())
+	return Json::Value(Json::nullValue);
```

Comment:
I'm not sure I understand the issue with serializing `null`. My idea behind `null` is that (of course depending on the type-safety of the language used after having parsed the json) you cannot really add `null` to a finite gas value, whereas adding `-1` to a finite gas value yields an actual result.

On the other hand, gas values might be quite large - this is an estimation after all and we use bigints here. So perhaps decimal-encoded integers for finite values and the string `infinite` for unknown upper bounds would be the best solution here.